src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1074,12 +1074,11 @@ private ImmutableArray<Diagnostic> FilterDiagnosticsSuppressedInSourceOrByAnalyz
             Compilation compilation,
             SuppressMessageAttributeState suppressMessageState)
         {
-            if (diagnostics.IsEmpty)
+            if (diagnostics.IsEmpty || compilation.Options.ReportSuppressedDiagnostics)
             {
                 return diagnostics;
             }
 
-            var reportSuppressedDiagnostics = compilation.Options.ReportSuppressedDiagnostics;
             var builder = ImmutableArray.CreateBuilder<Diagnostic>();
```

Comment:
If you want to reduce allocations here, I think you can instead use ArrayBuilder and ToImmutableAndFree in this method.

Reply:
If this code is source-included in some other project it still may not work, but could be worth a try.